tests/download.bats

```diff
@@ -46,3 +56,49 @@ EOM
   assert_line --regexp "https://github.com/cultureamp/ecr-scan-results-buildkite-plugin/releases/latest/download/ecr-scan-results-buildkite-plugin_linux_amd64 ecr-scan-results-buildkite-plugin"
   assert_line --regexp "executing ecr-scan-results-buildkite-plugin"
 }
+
+@test "Attempts to download and succeed after 2 tries" {
+
+  # Fails for first 2 tries and passes on later 
+
+  stub curl \
+      "echo 'curl 1 - Could not resolve host' && exit 6" \
+      "echo 'curl 2 - Failed to connect to example.com port 80 - Connection refused' && exit 7" \
+      "echo 'curl 3 - Success' && exit 0"
+
+  stub sleep "${sleep_stubs[@]}"
+  is_stubbed=true
+
+  # executes the retry download command
+  run retry_download "https://example.com" "$TMPDIR/output-plugin" "curl"
+
+  
+  assert_success
+  assert_line --partial "Attempt 1"
+  assert_line --partial "Attempt 2"
+  assert_line --partial "Success"
+
+}
+
+@test "Attempts to download and fails after maximum tries" {
+
+  # Fails for all 5 (maximum) tries 
+
+  stub curl \
+        "echo 'curl 1 - Could not resolve host: example.com' && exit 6" \
+        "echo 'curl 2 (7) Failed to connect to example.com port 80 - Connection refused' && exit 7" \
+        "echo 'curl 3 (7) Failed to connect to example.com port 80 - Connection refused' && exit 7"  \
+        "echo 'curl 4 (7) Failed to connect to example.com port 80 - Connection refused' && exit 7"  \
+        "echo 'curl 5 (7) Failed to connect to example.com port 80 - Connection refused' && exit 7"  \
+        "echo 'curl 6 (7) Failed to connect to example.com port 80 - Connection refused' && exit 7"  \
```

Comment:
Invalid trailing `\` escape.

Also, is this the correct number of items in the plan? Seems like it's one too many.